### DISCUSSION TITLE
Subscribe API Parameters.

### DISCUSSION THREAD

**Discussion by 2316Meheraj** [ONDC-Official/ONDC-FIS-Specifications#40] [Q&A]
Hi @92shreyansh ,
Need clarity in callback_url,subscriber_url,domain,type receiving following error:
{"message":{"ack":{"status":"NACK"}},"error":{"type":"CONTEXT-ERROR","code":"151","path":null,"message":
"Please provide valid schema. ERROR : [message.entity.callback_url: is required or Invalid., 
message.network_participant[0].subscriber_url: is required or Invalid.,
 message.network_participant[0].domain: is required or Invalid., 
 message.network_participant[0].type: is required or Invalid.]"}}

{
    "context": {
        "operation": {
            "ops_no": 1
        }
    },
    "message": {
        "request_id": "EASYPY-01923",
        "timestamp": "2023-09-4T12:55:54.101Z",
        "entity": {
            "gst": {
                "legal_entity_name": "EasyPay Private Limited",
                "business_address": "2nd Floor, TOWER-C, Tech Park One, Panchshil, Yerawada, Pune, Maharashtra 411006",
                "city_code": [
                    "std:080"
                ],
                "gst_no": "27AADCE6257B1ZU"
            },
            "pan": {
                "name_as_per_pan": "EASY PAY PRIVATE LIMITED",
                "pan_no": "ASDFP7657Q",
                "date_of_incorporation": "27/02/2014"
            },
            "name_of_authorised_signatory": "PRATEEK P KATYAL",
            "email_id": "prateek@easypay.in",
            "mobile_no": 9717539999,
            "country": "IND",
            "subscriber_id": "ondcuat.easypay.co.in",
            "unique_key_id": "636",
            "callback_url": "https://ondcuat.easypay.co.in/ecommerce/ondc/buyer/",
            "key_pair": {
                "signing_public_key": "3yP1Wvi/hPaimEAYJZ8C2BWC59D4DmFozg3KkuMYeDU=",
                "encryption_public_key": "MCowBQYDK2VuAyEAduMuZgmtpjdCuxv+Nc49K0cB6tL/Dj3HZetvVN7ZekM=",
                "valid_from": "2023-09-04T16:00:19.821Z",
                "valid_until": "2025-09-04T16:12:19.821Z"
            }
        },
        "network_participant": [
            {
                "subscriber_url": "https://ondcuat.easypay.co.in/ecommerce/ondc/buyer/",
                "domain": "ONDC:FIS12",
                "type": "BAP",
                "msn": false,
                "city_code": [
                    "std:020"
                ]
            }
        ]
    }
}

**Comment by nemishzalavadiya-pirimid** [comment]
{
  "context": {
    "operation": {
      "ops_no": 1
    }
  },
  "message": {
    "request_id": "27baa06d-f90a-486c-85e5-cc621b787f04",
    "timestamp": "2022-07-08T13:44:54.101Z",
    "entity": {
      "gst": {
        "legal_entity_name": "ABC Incorporates",
        "business_address": "Trade World, Mansarpur, Coorg, Karnataka 333333",
        "city_code": [
          "std:080"
        ],
        "gst_no": "07AAACN2082N4Z7"
      },
      "pan": {
        "name_as_per_pan": "ABC Incorporates",
        "pan_no": "ASDFP7657Q",
        "date_of_incorporation": "23/06/1982"
      },
      "name_of_authorised_signatory": "Anand Sharma",
      "address_of_authorised Signatory": "405, Pinnacle House, Kandiwali, Mumbai 400001",
      "email_id": "anand.sharma@abc.com",
      "mobile_no": 9912332199,
      "country": "IND",
      "subscriber_id": "sit.grab.in",
      "unique_key_id": "27baa06d-f90a-486c-85e5-cc621b787f04",
      "callback_url": "/ondc/onboarding",
      "key_pair": {
        "signing_public_key": "QSax2KT4UiTUWUqoVUaEcWhBcGTTNu+Sf8EMDRY1GaE=",
        "encryption_public_key": "O74ukMymk4KZnVs3sZhU2U7RXpaZ/qiOUMk5NWt6rbI=",
        "valid_from": "2022-07-08T13:44:54.101Z",
        "valid_until": "2022-07-08T13:44:54.101Z"
      }
    },
    "network_participant": [
      {
        "subscriber_url": "/bapl",
        "domain": "nic2004:52110",
        "type": "buyerApp",
        "msn": false,
        "city_code": [
          "std:080"
        ]
      }
    ]
  }
}

what is `subscriber_url` here?

**Comment by 92shreyansh** [comment]
callback_url - This is a relative url i.e. `https://<subsciber_id>/<callback_url>/on_subscribe` where you would be receiving the on_subscibe request for subscription
subscriber_url - This is releative url i.e `https://<subsciber_id>/<subsciber_url>/<search/on_search/select/on_select etc.>` where you would be reciving the transaction requests
domain - ONDC domain that you want to enable i.e. ONDC:FIS12(Credit), ONDC:TRV10(Ride Hailing) etc.
type - This is to identify buyerApp or seller App

**Comment by 92shreyansh** [comment]
One request to @2316Meheraj  or @nemishzalavadiya-pirimid  It would be great if you can move this thread with all these details and response to developer docs repo

**Reply by Shaikh-Dilshad** [reply]
Hey Shreyansh can you help me with my on subscription step please if possible

**Comment by ps1405** [comment]
Hi all,
i receiving following errror:
message.entity.key_pair.valid_from: is required or Invalid., message.entity.key_pair.valid_until: is required or Invalid.
{
    "endpoint_used": "https://preprod.registry.ondc.org/subscribe",
    "status": 200,
    "curl_error": null,
    "request": {
        "context": {
            "operation": {
                "ops_no": 1
            }
        },
        "message": {
            "request_id": "de3399ce-66c9-4096-887a-1a5aa8b42d64",
            "timestamp": "2025-11-05T10:37:01Z",
            "entity": {
                "gst": {
                    "legal_entity_name": "Ausha Digital Pvt Ltd",
                    "business_address": "2nd Floor, X 2119, Sector 25, Akshar Business Park, Vashi, Navi Mumbai 400703",
                    "city_code": [
                        "std:022"
                    ],
                    "gst_no": "27AAOCA6158Q1ZS"
                },
                "pan": {
                    "name_as_per_pan": "AUSHA DIGITAL PRIVATE LIMITED",
                    "pan_no": "AAOCA6158Q",
                    "date_of_incorporation": "2020-01-01"
                },
                "name_of_authorised_signatory": "Bhushan Patil",
                "address_of_authorised_signatory": "Vashi",
                "email_id": "b@ausha.in",
                "mobile_no": "9819912707",
                "country": "IND",
                "subscriber_id": "spoctree.ausha.dev",
                "unique_key_id": "525b3f02-2ad8-4b1a-bb8a-4d0eb34e4fd5",
                "callback_url": "/ondc_c/on_subscribe",
                "key_pair": {
                    "signing_public_key": "M/TTb4pCa9POFOUzEyGsTZpUnFO2+nQ5IiWwXRVlYw4=",
                    "encryption_public_key": "MCowBQYDK2VuAyEABPrWkucEVKDWtUzfXrXgs9xLq04yN/iMbh0OA/aXW3A=",
                    "valid_from": "2025-11-05T10:37:01Z",
                    "valid_until": "2026-11-05T10:37:01Z"
                }
            },
            "network_participant": [
                {
                    "subscriber_url": "/ondc",
                    "domain": "nic2004:60232",
                    "type": "buyerApp",
                    "msn": false,
                    "city_code": [
                        "std:022"
                    ]
                }
            ]
        }
    },
    "response": {
        "message": {
            "ack": {
                "status": "NACK"
            }
        },
        "error": {
            "type": "CONTEXT-ERROR",
            "code": "151",
            "path": null,
            "message": "Please provide valid schema. ERROR : [message.entity.key_pair.valid_from: is required or Invalid., message.entity.key_pair.valid_until: is required or Invalid.]"
        }
    }
}
